### PR TITLE
Rangemacro: add more tests

### PR DESCRIPTION
**Pull request by grebe**
@chick @ducky64 , I added some more basic tests

### CODE REVIEW

File: src/test/scala/chiselTests/RangeSpec.scala [hunk 1]
```diff
@@ -8,10 +8,64 @@ import org.scalatest.{Matchers, FreeSpec}
 class RangeSpec extends FreeSpec with Matchers {
   "Ranges can be specified for UInt, SInt, and FixedPoint" - {
     "to specify a UInt" in {
-      val x = UInt(range"[0, 7)")
+      val x = UInt(range"[0, 8)")
       x.getWidth should be (3)
 
       println(range"[4,32)")
```
Comment (ducky64) [inline src/test/scala/chiselTests/RangeSpec.scala]: Consider getting rid of prinlns in tests unless there's no other way to verify the results.

Comment (ducky64) [inline src/test/scala/chiselTests/RangeSpec.scala]: A better way might be to check equality to the raw tuple generated from the range macro.

File: src/test/scala/chiselTests/RangeSpec.scala [hunk 2]
```diff
@@ -8,10 +8,64 @@ import org.scalatest.{Matchers, FreeSpec}
 class RangeSpec extends FreeSpec with Matchers {
   "Ranges can be specified for UInt, SInt, and FixedPoint" - {
     "to specify a UInt" in {
-      val x = UInt(range"[0, 7)")
+      val x = UInt(range"[0, 8)")
```
Comment (ducky64) [inline src/test/scala/chiselTests/RangeSpec.scala]: Do you want to check edge case [0, 0] in here? Or is the behavior still under debate?

Comment (grebe) [inline src/test/scala/chiselTests/RangeSpec.scala]: I did check it, but for some reason put it in the test that was checking for illegal behavior. I moved it

File: src/test/scala/chiselTests/RangeSpec.scala (hunk 1 above)
Comment (grebe) [inline src/test/scala/chiselTests/RangeSpec.scala]: I'll add a test that checks the macros by themselves, without factory methods

**Comment by ducky64** [comment]
Do you also want to test interpolated variables, both literals (like range"[0, ${1+1}]" and variables?

**Comment by grebe** [comment]
Is there a way to test invalid macro invocations? I.e. range"(-1 1)" should be illegal because it is missing a comma.

**Comment by ducky64** [comment]
I don't think it's possible to test invalid macro invocations, since those would fail at compile time, before the test suite even starts running.

**Comment by ducky64** [comment]
Also, LGTM, merging.

**Comment by sdtwigg** [comment]
ScalaTest assertTypeError is designed to test if code fails to typecheck
(supplied code must still be parsable).

It is commonly used to check macro invocations and typelevel things.

**Comment by ducky64** [comment]
Cool, thanks! Added some compile-time failure checks. 04a1d71
